crates/bevy_sprite/src/mesh2d/material.rs

```diff
@@ -165,30 +264,48 @@ pub enum AlphaMode2d {
 
 /// Adds the necessary ECS resources and render logic to enable rendering entities using the given [`Material2d`]
 /// asset type (which includes [`Material2d`] types).
-pub struct Material2dPlugin<M: Material2d>(PhantomData<M>);
+///
+/// If `MESH_MATERIAL` is set to `true`, material handles will be retrieved from [`MeshMaterial2d`] components.
+/// Otherwise, they will be retrieved by querying for [`Handle<M>`] directly.
+pub struct Material2dPlugin<M: Material2d, const MESH_MATERIAL: bool = true>(PhantomData<M>);
 
-impl<M: Material2d> Default for Material2dPlugin<M> {
+impl<M: Material2d, const MESH_MATERIAL: bool> Default for Material2dPlugin<M, MESH_MATERIAL> {
     fn default() -> Self {
         Self(Default::default())
     }
 }
 
-impl<M: Material2d> Plugin for Material2dPlugin<M>
+impl<M: Material2d, const MESH_MATERIAL: bool> Plugin for Material2dPlugin<M, MESH_MATERIAL>
 where
     M::Data: PartialEq + Eq + Hash + Clone,
 {
     fn build(&self, app: &mut App) {
         app.init_asset::<M>()
+            .register_type::<MeshMaterial2d<M>>()
+            .register_type::<HasMaterial2d>()
             .add_plugins(RenderAssetPlugin::<PreparedMaterial2d<M>>::default());
 
         if let Some(render_app) = app.get_sub_app_mut(RenderApp) {
+            render_app.add_systems(ExtractSchedule, clear_material_2d_instances::<M>);
+
+            if MESH_MATERIAL {
+                render_app.add_systems(
+                    ExtractSchedule,
+                    extract_mesh_materials_2d::<M>.after(clear_material_2d_instances::<M>),
+                );
+            } else {
+                render_app.add_systems(
+                    ExtractSchedule,
+                    extract_materials_2d::<M>.after(clear_material_2d_instances::<M>),
+                );
+            }
+
```

Comment:
I'm reverting a lot of this in a bit, I realized I can of course just make wireframes use `MeshMaterial2d` as well and get rid of this special-casing :P